src/compiler/parser.ts

```diff
@@ -4545,7 +4545,18 @@ namespace ts {
                         parseErrorAtRange(openingTag, Diagnostics.JSX_fragment_has_no_corresponding_closing_tag);
                     }
                     else {
-                        parseErrorAtRange(openingTag.tagName, Diagnostics.JSX_element_0_has_no_corresponding_closing_tag, getTextOfNodeFromSourceText(sourceText, openingTag.tagName));
+                        // We want the error to span only property identifier "Foo.Bar" e.g in < Foo.Bar >...
+                        const tag = openingTag.tagName as any;

```

Comment:
Why is this `as any` necessary?

Reply:
The compiler was complaining about missing properties on `tagName`. I tried using some types from the types.ts file but that only partially solved the warnings.